arm-notificationhubs/2016-03-01/swagger/notificationhubs.json

```diff
@@ -1819,7 +1819,6 @@
           "description": "Resource Id"
         },
         "name": {
-          "readOnly": true,
```

Comment:
@amolr - Another way to isolate this problem is to define all the properties for the body parameter of checkNamespaceAvailability and it should not inherit from Resource. Then you can keep name as a "readonly" property on Resource.